crazy-filters/ui/image_transform.py

```diff
@@ -106,7 +106,9 @@ def get_frame(self):
 
     def apply_all_transforms(self):
         buf = self.get_frame()
-        for image_transform, kwargs in self.transforms:
+        for image_transform, kwargs in self.transforms.items():
+            if 'ticking' in kwargs:
+                kwargs['ticking'][0] += 1

```

Comment:
Maybe it could be considered to use a more specific object for that like:

```
class carrousel_context:
  def __init__(self):
    self.tick = 0
  def next_apply(self):
    self.tick += 1
```

You could then have:
`self.add_button('Carrousel', tf.carrousel_transfo, filter_context=carousel_context())`

And in the current line:
if "filter_context' in kwargs:
  filter_context.next_apply()

=> A bit more verbose but maybe more explicit and extensible that storing the int in a list. -> it delegates the implementation of what happens for carousel's ticks to something more specific (and thus easier to evolve) and at the same time it introduces a contract (the next_apply method and the filter_context option name for other filters if they need so)